crates/downloader/src/main.rs

```diff
@@ -524,6 +537,114 @@ fn find_onnxruntime(
         .with_context(|| "指定されたOS, アーキテクチャ, デバイスを含むものが見つかりませんでした")
 }
 
+/// ダウンロードすべきモデル、利用規約を見つける。その際ユーザーに利用規約の同意を求める。
+async fn find_models(octocrab: &Octocrab, repo: &RepoName) -> anyhow::Result<ModelsWithTerms> {
+    let repos = octocrab.repos(&repo.owner, &repo.repo);
+
+    let (tag, sha) = repos
+        .list_tags()
+        .send()
+        .await?
+        .into_iter()
+        .map(
+            |Tag {
+                 name,
+                 commit: CommitObject { sha, .. },
+                 ..
+             }| {
+                let tag = name
+                    .parse()
+                    .with_context(|| format!("`{repo}` contains non-SemVer tags"))?;
+                Ok((tag, sha))
+            },
+        )
+        .collect::<anyhow::Result<Vec<_>>>()?
+        .into_iter()
+        .filter(|(version, _)| ALLOWED_MODELS_VERSIONS.matches(version))
+        .sorted()
+        .last()
+        .with_context(|| format!("`{repo}`"))?;
+    let tag = tag.to_string();
+
+    let terms = repos
+        .get_content()
+        .r#ref(&sha)
+        .path(format!("{MODELS_DIR_NAME}/{MODELS_TERMS_FILE}"))
+        .send()
+        .await?
+        .items
+        .into_iter()
+        .exactly_one()
+        .map_err(|_| anyhow!("could not find `{MODELS_TERMS_FILE}`"))?;
+    ensure!(
+        terms.encoding.as_deref() == Some("base64"),
+        r#"expected `encoding="base64"`"#,
+    );
+    let terms = terms.content.expect("should present").replace('\n', "");
+    let terms = BASE64_STANDARD.decode(terms)?;
+    let terms = String::from_utf8(terms)
+        .with_context(|| format!("`{MODELS_TERMS_FILE}` is not valid UTF-8"))?;
+    ensure_confirmation(&terms, MODELS_TERMS_NAME)?;
+
+    let models = repos
+        .get_content()
+        .r#ref(&sha)
+        .path(MODELS_DIR_NAME)
+        .send()
+        .await?
+        .items
+        .into_iter()
+        .filter(|Content { name, r#type, .. }| name != MODELS_TERMS_FILE && r#type == "file")
+        .map(
+            |Content {
+                 name,
+                 size,
+                 download_url,
+                 ..
+             }| GhContent {
+                name,
+                download_url: download_url.expect("should present"),
+                size: size as _,
+            },
+        )
+        .collect();
+
+    Ok(ModelsWithTerms { tag, terms, models })
+}
+
+fn ensure_confirmation(terms: &str, terms_name: &'static str) -> anyhow::Result<()> {
+    eprintln!(
+        "----------BEGIN {terms_name}----------\n\
+         {terms}----------END {terms_name}----------",
```

Comment:
細かいけど、頭に改行足すなら後ろに改行足すのがシンメトリーかも。
というのと、termsの最後で改行忘れることもありそうなので改行足しときたみ。
あんま強い意見じゃないです！

```suggestion
        "----------BEGIN {terms_name}----------\n\
         {terms}\n\
         ----------END {terms_name}----------",
```

Reply:
[`3f26512` (#928)](https://github.com/VOICEVOX/voicevox_core/pull/928/commits/3f26512fc1f39a9ba06e27c72d1c2c29beadd3d5): ファイル末尾を正規化する、という方向にしてみました。